server/team/team.contoller.js

```diff
@@ -2,20 +2,111 @@ const debug = require('debug');
 
 const Team = require('./team.model');
 const User = require('../user/user.model');
+const launchChromeAndRunLighthouse = require('../lighthouse/lighthouse');
 
 const log = debug('fcc:team:controller');
 
 function create(req, res) {
   const { body: team } = req;
   log(team);
-  return Team.create(team)
+  Team.create(team)
     .then((newTeam) => {
       if (!newTeam) {
         return res.status(500).json({ acknowladged: false });
       }
-      res.status(200).json({ acknowladged: true });
-      return newTeam._id;
+      return Team.findOneAndUpdate(
+        {
+          _id: newTeam._id
+        },
+        {
+          $push: {
+            collaborators: req.user
+          }
+        }
+      )
+      .then((teamUpdate) => {
+        res.status(200).json({ acknowladged: true });
+        return teamUpdate._id;
+      });
     })
     .then(teamId => User.update({ _id: req.user._id }, { teamId }));
 }
-module.exports = { create };
+
+function join(req, res) {
+  Team.findOneAndUpdate(
+    { _id: req.body.team },
+    { $push: { collaborators: req.user } },
+    { safe: true, new: true, multi: false }
+  )
+  .then((team) => {
+    log(team);
+    return res.redirect(`/user/${req.user._id}`);
+  });
+}
+
+function joinForm(req, res) {
+  return res.render('joinTeam');
+}
+
+function analyze(req, res) {
```

Comment:
RE discussion on `gitter`, We shoudn't open up the lighthouse test to the internet, it should only be accessible via a webhook from Netlify